src/Microsoft.AspNet.OData.Shared/EdmStructuredObject.cs

```diff
@@ -224,8 +225,15 @@ internal static object GetDefaultValue(IEdmTypeReference propertyType)
                 if (propertyType.IsPrimitive() ||
                     (isCollection && propertyType.AsCollection().ElementType().IsPrimitive()))
                 {
+
+                    bool hasDefaultConstructor = (!clrType.IsClass) || clrType.GetConstructors(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static).
+                                                 Any(x => x.GetParameters().Count() == 0);
```

Comment:
`GetParameters` returns an array, so calling `Length` instead of `Count()` should save us a cast